From: Andrea Azzarone <andrea.azzarone@canonical.com>
Date: Tue, 31 Jul 2018 04:44:12 +0100
Subject: online-accounts: Hide window after adding an online account from
 another app

Some applications (e.g. gnome-calendar, gnome-todo, etc.) use gnome-control-center
to allow the user to add online accounts. This patch makes sure that g-c-c main window
is closed after the user interacted with the new-account dialog.

Bug-Ubuntu: https://launchpad.net/bugs/1754651
Forwarded: no
Last-Update: 2018-03-09
---
 panels/online-accounts/cc-online-accounts-panel.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/panels/online-accounts/cc-online-accounts-panel.c b/panels/online-accounts/cc-online-accounts-panel.c
index 2a626c2..b41fb32 100644
--- a/panels/online-accounts/cc-online-accounts-panel.c
+++ b/panels/online-accounts/cc-online-accounts-panel.c
@@ -799,7 +799,10 @@ cc_online_accounts_panel_set_property (GObject      *object,
             }
 
           if (g_strcmp0 (first_arg, "add") == 0)
-            command_add (CC_ONLINE_ACCOUNTS_PANEL (object), parameters);
+            {
+              command_add (CC_ONLINE_ACCOUNTS_PANEL (object), parameters);
+              g_application_quit (g_application_get_default ());
+            }
           else if (first_arg != NULL)
             select_account_by_id (CC_ONLINE_ACCOUNTS_PANEL (object), first_arg);
 
